synapse/handlers/presence.py

```diff
@@ -680,7 +706,15 @@ async def _update_states(self, new_states: Iterable[UserPresenceState]) -> None:
             if to_federation_ping:
                 federation_presence_out_counter.inc(len(to_federation_ping))
 
-                self._push_to_remotes(to_federation_ping.values())
+                hosts_and_states = await get_interested_remotes(
+                    self.store,
+                    self.presence_router,
+                    list(to_federation_ping.values()),
+                    self.state,
+                )
+
+                for destinations, states in hosts_and_states:
+                    self.federation.send_presence_to_destinations(states, destinations)
```

Comment:
Because `_persist_and_notify` will cause the presence to be replicated to the other instances, which will then send out the presence, while here we're resending the presence to particular remotes. i.e. there are two cases:

1. Presence state update, which gets streamed out via presence replication stream; and
2. Times when we want to send existing presence states to remotes, which happens a) when a host joins a room and we send out all local presence and b) we resend "online" presence periodically as a ping/keepalive mechanism.